wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -351,7 +419,9 @@ public NodeList getFileData(String logicalname, Long beginrow, Integer numrows,
                     DFUDataColumn[] thesecols = (DFUDataColumn[]) r;
                     if (r != null)
                     {
-                        cols.addAll(Arrays.asList(thesecols));
+                        for (DFUDataColumn col:Arrays.asList(thesecols)) {
```

Comment:
same thing in line 420, if (thesecols != null), it's equivalent, but we're actually referencing thesecols in the loop